File: tests/drivers/node/v5_8/read_routing.js
const neo4j = require('neo4j-driver');

function die() {
  session.close();
  driver.close();
  process.exit(1);
}

function Neo4jService(uri) {
  const driver = neo4j.driver(uri, neo4j.auth.basic("", ""));

  async function readGreeting() {
    const session = driver.session({ defaultAccessMode: neo4j.session.READ });
    try {
      const result = await session.readTransaction(tx =>
        tx.run('MATCH (n:Greeting) RETURN n.message AS message')
      );
      console.log("Read txn finished");
    } finally {
      await session.close();
    }
  }

  async function close() {
    await driver.close();
  }

  return {
    readGreeting,
    close
  };
}

async function readGreetingsFromUri(uri) {
  const service = Neo4jService(uri);
  await service.readGreeting();
  await service.close();
}

async function main() {
  console.log("Started reading route");
  const uris = [
    'neo4j://localhost:7690',
    'neo4j://localhost:7691',
    'neo4j://localhost:7692'
  ];

  try {
    for (const uri of uris) {
      await readGreetingsFromUri(uri);
    }
  } catch (error) {
    console.error('An error occurred:', error);
    die();
  }
  console.log("Finished reading route");
}

main().catch(error => console.error(error));

File: tests/drivers/python/v5_8/read_routing.py
from neo4j import GraphDatabase


class Neo4jService:
    def __init__(self, uri, user="", password=""):
        self.driver = GraphDatabase.driver(uri, auth=(user, password))

    def close(self):
        self.driver.close()

    def read_greeting(self):
        with self.driver.session() as session:
            session.execute_read(self._create_and_return_greeting)
            print("Read txn passed!")

    @staticmethod
    def _create_and_return_greeting(tx):
        tx.run("MATCH (n:Greeting) RETURN n.message AS message")


def read_greetings_from_uri(uri):
    service = Neo4jService(uri)
    service.read_greeting()
    service.close()


def main():
    print("Started reading route")
    uris = ["neo4j://localhost:7690", "neo4j://localhost:7691", "neo4j://localhost:7692"]

    try:
        for uri in uris:
            read_greetings_from_uri(uri)
    except Exception as error:
        print(f"An error occurred: {error}")
        exit(-1)
    print("Finished reading route")


if __name__ == "__main__":
    main()
